cabal-install/Distribution/Client/Install.hs

```diff
@@ -597,6 +597,13 @@ checkPrintPlan verbosity installed installPlan sourcePkgDb
     dryRun            = fromFlag (installDryRun            installFlags)
     overrideReinstall = fromFlag (installOverrideReinstall installFlags)
 
+    -- Escalate verbosity until level has reached at least 'verbose'
+    -- NB: this makes sure we don't lose any verbosity flags
+    atLeastVerbose :: Verbosity -> Verbosity
```

Comment:
I'm confused. Why doesn't ``verbosity `max` verbose`` work?

Reply:
because neither

```hs
isVerboseXXX verbosity == isVerboseXXX (verbosity `max` verbose)
```
nor

```hs
isVerboseXXX verbosity == isVerboseXXX (verbose `max` verbosity)
```

(where XXX stands for one of the verbosity flags) holds for all possible values of `verbosity`.

Reply:
Ah right, forgot that `Verbosity` has additional state now.